reproducibility_project/src/analysis/sampler.py

```diff
@@ -42,6 +44,46 @@ def sample_job(
     job.doc["sampling_results"][variable] = (range(start, stop, step), Neff)
 
 
+def write_subsampled_values(
+    job: signac.contrib.project.Job,
+    property: str,
+    property_filename: str = "log.txt",
+    overwrite: bool = False,
+) -> None:
+    """Write out subsampled values to a Job's 'data' document."""
+    if not isinstance(job, signac.contrib.project.Job):
+        raise TypeError(
+            f"Expected input 'job' of type signac.contrib.project.Job, was provided: {type(job)}"
+        )
+
+    if property is None or property == "":
+        raise ValueError(
+            f"Expected 'property' to be a name of a property, was provided {property}."
+        )
+
+    if (
+        not overwrite
+        and job.data.get(f"subsamples/{property}", None) is not None
+    ):
+        raise ValueError(
+            f"Attempting to overwrite already existing data for property: {property}, set overwrite=True to do this."
+        )
+
+    sampling_indices = job.doc["sampling_results"][f"{property}"]
```

Comment:
Note: We cannot serialize a python `range` object to json. So we need to also update `write_subsampled_values` above

```suggestion
    sampler_output= job.doc["sampling_results"][f"{property}"]
```